src/Opaleye/Operators.hs

```diff
@@ -469,6 +473,27 @@ timestamptzAtTimeZone = C.binOp HPQ.OpAtTimeZone
 dateOfTimestamp :: F.Field T.SqlTimestamp -> F.Field T.SqlDate
 dateOfTimestamp (Column e) = Column (HPQ.FunExpr "date" [e])
 
+-- | @IntervalNum from to@ determines from which date or time types an interval
+-- can be added ('addInterval') or subtracted ('minusInterval`) and which is the
+-- resulting type.
+--
+-- The instances should correspond to the interval + and - operations listed in:
+--
+-- https://www.postgresql.org/docs/current/functions-datetime.html#OPERATORS-DATETIME-TABLE
+class IntervalNum from to
```

Comment:
Would it be sensible to have a functional dependency here?

Reply:
Initially I thought we could have arbitrary combinations but thinking about it again I think it makes sense to add a fundep here. Will do.